src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -110,5 +108,31 @@ internal static bool CompletionRequiresQuotes(string completion, bool escape)
 
         private static bool ContainsCharsToCheck(ReadOnlySpan<char> text, bool escape)
             => text.ContainsAny(escape ? s_escapeCharsToCheck : s_defaultCharsToCheck);
+
+        internal static string QuoteCompletionText(
```

Comment:
Please remember why do we need the method? What issues we address? Why don't we add new tests for the issues?

Reply:
So before surrounding completion text with quotes, it makes sure chars are escaped. You can see very similar logic from https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs#L237-L258.

Instead of repeating this logic is seems much better to wrap in a reusable method `QuoteCompletionText`. We could potentially replace many parts of code with this method.

Reply:
> Instead of repeating this logic is seems much better to wrap in a reusable method `QuoteCompletionText`.

In the case we need to use the method in the PR.

Reply:
I have replaced simple method calls with this method after adding some more escape parameters to QuoteCompletionText. Will need to add tests now.